run.py

```diff
@@ -63,11 +51,32 @@ def _environment(self):
         result = {}
         result.update(os.environ)
         result['PROTOCOL_VERSION'] = '3'
```

Comment:
this is not good we need to test all protocol versions not only '3' - I think that by default we limited it to '3' since we did not support 4 but now we do support 4 and should test it as well - can be another pull_request,

If there is something that is not working with '4'  we need to understand why